### ISSUE TITLE
ecurve/bigi vs bn.js - short-term/long-term plans

### ISSUE THREAD

**Issue by runn1ng** [bitcoinjs/bitcoinjs-lib#610]
I have looked at bigi and ecurve githubs and they seem to be semi-deprecated in favor of bn.js library.

What is the plan for bitcoinjs-lib. Do you have plans to stay on ecurve+bigi, or do you plan to migrate to bn.js? And in what approximate timeframe?

**Comment by dcousens** [comment]
@runn1ng I'm sorry I hadn't replied to your email about this sooner.

The plan changes depending on general consensus and review,  however,  it appears to be currently approximating the full removal of `ecurve` and `bigi` in support of `bn.js` and `elliptic`.

The timeframe would be for `3.0.0`, as too much of the API has interchange with the `ecurve` and `bigi` primitives.

The timeline for `3.0.0`,  while unclear at this time,  I'd say it would be before 2017.
See https://github.com/bitcoinjs/bitcoinjs-lib/milestones for things remaining to do before we can move on,  segregated witness will certainly take priority.

**Comment by runn1ng** [comment]
(I was thinking maybe the e-mail didn't arrive so it would be better to ask here :) and it's more open anyway)

**Comment by jprichardson** [comment]
I can say this much, it's my intention to start a 3.0 branch with [secp256k1](https://github.com/cryptocoinjs/secp256k1-node) (also includes elliptic/bn.js). I'd be fine with 3.0 just including that change and then focusing on 4.0 for bigger changes.

**Comment by dcousens** [comment]
@jprichardson works for me.

**Comment by dcousens** [comment]
Closing in favour of https://github.com/bitcoinjs/bitcoinjs-lib/pull/623
